streams/src/main/java/org/apache/kafka/streams/kstream/internals/KGroupedTableImpl.java

```diff
@@ -132,18 +131,18 @@ private void determineIsQueryable(final String queryableStoreName) {
         ChangedDeserializer<? extends V> changedValueDeserializer = new ChangedDeserializer<>(valueDeserializer);
 
         // send the aggregate key-value pairs to the intermediate topic for partitioning
-        topology.addInternalTopic(topic);
-        topology.addSink(sinkName, topic, keySerializer, changedValueSerializer, this.name);
+        builder.internalTopologyBuilder.addInternalTopic(topic);
```

Comment:
I did some of the refactoring you suggested already. But it's easier and cleaner, to go from `Topology` to `InternalTopologyBuilder` at one place. For internals, we don't loose anything if we use the internal topology builder IMHO.